qualtran/surface_code/data_block.py

```diff
@@ -14,113 +14,170 @@
+    state distillation.
     """
 
+    @property
     @abc.abstractmethod
-    def footprint(self, n_algo_qubits: int) -> int:
-        """The number of physical qubits used by the data block.
+    def data_d(self):
+        """The code distance used to store the data in the data block."""
+
+    @abc.abstractmethod
+    def n_tiles(self, n_algo_qubits: int) -> int:
+        """The number of surface code tiles used to store a given number of algorithm qubits.
+
+         We define an “algorithm qubit” to be a qubit used in the routing of algorithm-relevant
+         quantum data in a bloq. A physical qubit is a physical system that can encode one qubit,
+         albeit noisily. Specific to the surface code, we define a “tile” to be the minimal area
+         of physical qubits necessary to encode one logical qubit to a particular code distance.
+         A tile can store an algorithm qubit, can be used for ancillary purposes like routing,
+         or can be left idle. A tile is usually a square grid of $2d^2$ physical qubits.
 
-        Attributes:
-            n_algo_qubits: The number of algorithm qubits whose data must be stored and
-                accessed.
+         DataBlock implementations must override this method. This method is used by
+         `self.n_phys_qubits` to report the total number of physical qubits.
+
+        Args:
+            n_algo_qubits: The number of algorithm qubits to compute the number of tiles for.
+
+        Returns:
+            The number of tiles used by this data block to store the given number of algorithm
+            qubits.
         """
 
+    @property
     @abc.abstractmethod
-    def data_error(self, n_algo_qubits: int, n_cycles: int, phys_err: float) -> float:
-        """The error associated with storing data on `n_algo_qubits` for `n_cycles`."""
+    def n_steps_to_consume_a_magic_state(self):
+        """The number of surface code steps to consume a magic state.
 
-    @abc.abstractmethod
-    def n_cycles_to_consume_a_magic_state(self) -> int:
-        """The worst case number of cycles needed to consume a magic state."""
+        We must teleport in "magic states" to do non-Clifford operations on our algorithmic
+        data qubits. The layout of the data block can limit the number magic states consumed
+        per unit time.
+
+        One surface code step is `data_d` cycles of error correction.
+
+        DataBlock imlpementation must override this method. This method is used by
+        `self.n_cycles` to report the total number of cycles required.
+        """
+
+    def n_cycles(self, n_logi_gates: 'MagicCount', logi_err_model: 'LogicalErrorModel') -> int:
+        """The number of surface code cycles to apply the number of gates to the data block.
+
+        Note that only the Litinski (2019) derived data blocks model a limit on the number of
+        magic states consumed per step. Other data blocks return "zero" for the number of cycles
```

Comment:
> "zero"

I don't think this is true ... the CCZ2T doesn't talk about this and focuses on routing and distillation as the dominant factors but it also prefly mentions the work needed to teleport the CCZ magic state without specifying how long it takes in page 11

Reply:
The distinction is if you have e.g. 50 factories in a litinski data block you're limited by the number of t states you can consume. If you're using a gidney model, you are not. They pay heed to the time it takes to transport the magic state, yes, but assume that that can be intelligently routed and parallelized so the magic state throughput (not time/latency) is unbounded. 

With these models, there isn't "right" or "wrong" but rather fidelity to the original model, and I think in this case assuming unbounded t state consumption is closer to the original model than the restrictive 1-state-per-step. The gidney spreadsheet (in the supplementary materials) doesn't ever consider the data block being the time-limiting factor

Reply:
they way I have reconcilled these is by taking the max between generation and consumption cycles ... but I'm okay with saying it is zero for now

Reply:
the "identity operation" for max is 0. The current behavior, if you just want the generation cycle  because you're doing gidney form the max might still choose the consumption as the limiting factor